### PR TITLE
Fix custom margin on Layer component and refactor margin calculation

### PR DESCRIPTION
**Pull request by oorestisime**
Closes: #2456

Signed-off-by: Orestis Ioannou <oorestisime@gmail.com>

#### What does this PR do?

It all started from trying to fix the issue and then lead to a small refactoring

#### Where should the reviewer start?

Unit tests are passing, the one modifying was a part of the fix to the bug. I think i improved the `MARGINS` sugar. i am open to suggestions to further improve. It still seems to me a bit code-repeating but i prefer incremental improvements here.

#### What testing has been done on this PR?

Added a test for the custom margin

#### How should this be manually tested?

i was playing around in the storybook for this. Maybe this could help

```
const CustomLayer = () => (
<Grommet theme={grommet}>
  <Box background="dark-3">
    <Heading>body</Heading>
    <Layer background="dark-1" margin={{ top: '50px' }}>
      <Box pad="xlarge">text</Box>
      <Box pad="xlarge">text</Box>
      <Box pad="xlarge">text</Box>
    </Layer>
  </Box>
</Grommet>
);

```

#### Do the grommet docs need to be updated?
nope
#### Is this change backwards compatible or is it a breaking change?
compatible

### CODE REVIEW

File: src/js/components/Layer/StyledLayer.js [hunk 1]
```diff
@@ -69,15 +69,32 @@ export const StyledOverlay = styled.div`
...
-  top: (margin, theme) =>
-    theme.global.edgeSize[margin.top || margin.vertical || margin] || '0px',
-  bottom: (margin, theme) =>
-    theme.global.edgeSize[margin.bottom || margin.vertical || margin] || '0px',
-  left: (margin, theme) =>
-    theme.global.edgeSize[margin.left || margin.horizontal || margin] || '0px',
-  right: (margin, theme) =>
-    theme.global.edgeSize[margin.right || margin.horizontal || margin] || '0px',
+const getMargin = (margin, theme, position) => {
+  const axe =
```
Comment (alansouzati) [inline src/js/components/Layer/StyledLayer.js]: axis ?

File: src/js/components/Layer/StyledLayer.js [hunk 2]
```diff
@@ -69,15 +69,32 @@ export const StyledOverlay = styled.div`
...
+    position === 'top' || position === 'bottom' ? 'vertical' : 'horizontal';
+  const marginApplied =
+    theme.global.edgeSize[margin[position] || margin[axe] || margin] ||
```
Comment (alansouzati) [inline src/js/components/Layer/StyledLayer.js]: can we do 

`const marginValue = margin[position] || margin[axe] || margin;` ?

Comment (alansouzati) [reply src/js/components/Layer/StyledLayer.js]: the .  `theme.global.edgeSize[marginValue] || marginValue`;

File: src/js/components/Layer/StyledLayer.js (hunk 1 above)
Comment (oorestisime) [reply src/js/components/Layer/StyledLayer.js]: oups french got in my head :D

**Pull request merged**